library/ptpulse/microphone.py

```diff
@@ -253,14 +266,28 @@ def record():
         print("Error: pi-topPULSE is not initialised.")
         sys.exit()
 
-    if _thread_running == False:
+    if _thread_running == False and _continue_writing == False:
         _thread_running = True
         _continue_writing = True
         _recording_thread = Thread(group=None, target=_thread_method)
         _recording_thread.start()
     else:
         print("Microphone is already recording!")
 
+def record_async():
+    """Start recording on the pi-topPULSE microphone - return generator"""
```

Comment:
Try and make function descriptions clearer to someone without knowledge of the internals